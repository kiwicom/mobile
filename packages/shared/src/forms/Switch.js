// @flow strict

import * as React from 'react';
import { Switch as OriginalSwitch } from 'react-native'; // eslint-disable-line no-restricted-imports
import { defaultTokens } from '@kiwicom/mobile-orbit';

type Props = {|
  +value: boolean,
  +onValueChange: (value: boolean) => void,
|};

export default function Switch(props: Props) {
  return (
    <OriginalSwitch
      onValueChange={props.onValueChange}
      value={props.value}
      onTintColor={defaultTokens.paletteProductNormal}
    />
  );
}
