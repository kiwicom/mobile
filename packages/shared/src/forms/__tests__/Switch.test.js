// @flow strict

import * as React from 'react';
import { PlaygroundRenderer } from '@kiwicom/mobile-playground';

import Switch from '../Switch';

it('renders with false', () => {
  PlaygroundRenderer.render(<Switch value={false} onValueChange={jest.fn()} />);
});

it('renders with true', () => {
  PlaygroundRenderer.render(<Switch value={true} onValueChange={jest.fn()} />);
});
