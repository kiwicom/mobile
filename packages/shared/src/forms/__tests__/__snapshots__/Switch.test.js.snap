// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with false 1`] = `
<Switch
  onValueChange={[MockFunction]}
  trackColor={
    Object {
      "true": "#00a991",
    }
  }
  value={false}
/>
`;

exports[`renders with true 1`] = `
<Switch
  onValueChange={[MockFunction]}
  trackColor={
    Object {
      "true": "#00a991",
    }
  }
  value={true}
/>
`;
